### PR TITLE
PR: Drop support for Python 2

### PR DESCRIPTION
**Pull request by impact27**
Fixes https://github.com/spyder-ide/spyder-kernels/issues/340

### CODE REVIEW

**Comment by ccordoba12** [comment]
@impact27, thanks a lot for your help with this! Let me know when it's ready for review.

**Comment by impact27** [comment]
@ccordoba12 I think this is read

File: spyder_kernels/customize/spydercustomize.py [hunk 1]
```diff
@@ -379,7 +323,7 @@ def get_current_file_name():
     try:
         return frontend_request().current_filename()
     except Exception:
-        _print("This command failed to be executed because an error occurred"
+        print("This command failed to be executed because an error occurred"
                " while trying to get the current file name from Spyder's"
                " editor. The error was:\n\n")
```
Comment (ccordoba12) [inline spyder_kernels/customize/spydercustomize.py:328]:
```suggestion
              " while trying to get the current file name from Spyder's"
              " editor. The error was:\n\n")
```

To align these lines correctly.

File: spyder_kernels/customize/spydercustomize.py [hunk 2]
```diff
@@ -640,25 +548,19 @@ def runcell(cellname, filename=None, post_mortem=False):
...
-        pass
     ipython_shell = get_ipython()
     try:
         # Get code from spyder
         cell_code = frontend_request().run_cell(cellname, filename)
     except Exception:
-        _print("This command failed to be executed because an error occurred"
+        print("This command failed to be executed because an error occurred"
                " while trying to get the cell code from Spyder's"
                " editor. The error was:\n\n")
```
Comment (ccordoba12) [inline spyder_kernels/customize/spydercustomize.py:558]:
```suggestion
              " while trying to get the cell code from Spyder's"
              " editor. The error was:\n\n")
```

Align lines.

File: spyder_kernels/customize/spyderpdb.py [hunk 3]
```diff
@@ -585,7 +563,7 @@ def _cmdloop(self):
                 self.allow_kbdint = False
                 break
             except KeyboardInterrupt:
-                _print("--KeyboardInterrupt--\n"
+                print("--KeyboardInterrupt--\n"
                        "For copying text while debugging, use Ctrl+Shift+C",
                        file=self.stdout)
```
Comment (ccordoba12) [inline spyder_kernels/customize/spyderpdb.py:568]:
```suggestion
                      "For copying text while debugging, use Ctrl+Shift+C",
                      file=self.stdout)
```

Align lines.

File: spyder_kernels/customize/umr.py [hunk 4]
```diff
@@ -139,5 +135,5 @@ def run(self):
         # Report reloaded modules
         if self.verbose and self.modnames_to_reload:
             modnames = self.modnames_to_reload
-            _print("\x1b[4;33m%s\x1b[24m%s\x1b[0m"
+            print("\x1b[4;33m%s\x1b[24m%s\x1b[0m"
                    % ("Reloaded modules", ": "+", ".join(modnames)))
```
Comment (ccordoba12) [inline spyder_kernels/customize/umr.py:139]:
```suggestion
                  % ("Reloaded modules", ": "+", ".join(modnames)))
```

Align line.

File: spyder_kernels/utils/iofuncs.py [hunk 5]
```diff
@@ -442,9 +437,9 @@ def setup(self):
                 save_extensions[filter_str] = ext
                 save_filters.append(filter_str)
                 save_funcs[ext] = savefunc
-        load_filters.insert(0, to_text_string("Supported files"+" (*"+\
+        load_filters.insert(0, str("Supported files"+" (*"+\
                                               " *".join(load_ext)+")"))
```
Comment (ccordoba12) [inline spyder_kernels/utils/iofuncs.py:441]:
```suggestion
        load_filters.insert(0, str("Supported files"+" (*"+\
                                  " *".join(load_ext)+")"))
```

Align line.

**Review by ccordoba12 — COMMENTED** [review]
Thanks @impact27 for your work on this! Let's try to get this merged soon so you don't have to fix more conflicts here (as you have now).

File: spyder_kernels/customize/spydercustomize.py [hunk 6]
```diff
@@ -518,29 +443,21 @@ def runfile(filename=None, args=None, wdir=None, namespace=None,
         if os.name == 'nt':
             filename = filename.replace('/', '\\')
 
-    try:
-        filename = filename.decode('utf-8')
-    except (UnicodeError, TypeError, AttributeError):
-        # UnicodeError, TypeError --> eventually raised in Python 2
-        # AttributeError --> systematically raised in Python 3
-        pass
```
[the remark this answers is not in the document]
Comment (impact27) [reply spyder_kernels/customize/spydercustomize.py:526]: AttributeError --> systematically raised in Python 3

File: spyder_kernels/customize/spydercustomize.py [hunk 7]
```diff
@@ -640,25 +548,19 @@ def runcell(cellname, filename=None, post_mortem=False):
         # Otherwise code caching doesn't work
         if os.name == 'nt':
             filename = filename.replace('/', '\\')
-    try:
-        filename = filename.decode('utf-8')
-    except (UnicodeError, TypeError, AttributeError):
-        # UnicodeError, TypeError --> eventually raised in Python 2
-        # AttributeError --> systematically raised in Python 3
-        pass
```
[the remark this answers is not in the document]
Comment (impact27) [reply spyder_kernels/customize/spydercustomize.py:648]: AttributeError --> systematically raised in Python 3

**Comment by ccordoba12** [comment]
@impact27, given that there are no more important changes for 2.3.0, and that's going to be the last big release of Spyder-kernels for Spyder 5, please try to finish this one so we can continue merging your other pending PRs here for Spyder-kernels 3/Spyder 6.

**Comment by impact27** [comment]
@ccordoba12 I think this is ready for review

File: spyder_kernels/utils/tests/test_dochelpers.py [hunk 8]
```diff
@@ -1,88 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (ccordoba12) [inline spyder_kernels/utils/tests/test_dochelpers.py:1]: Recover this test.

**Review by ccordoba12 — APPROVED** [review]
Thanks a lot @impact27 for your help with this!

**Pull request merged**